packages/@ionic/cli/src/commands/package/build.ts

```diff
@@ -335,24 +332,28 @@ if you do not wish to download ${input('apk')}.
       throw new Error(`Build ${build.state}`);
     }
 
-    for (const artifactType of artifactTypes) {
-      const url = await this.getDownloadUrl(appflowId, buildId, artifactType, token);
+    const availableArtifactTypes = build.artifacts.map((artifact) => artifact.artifact_type);
 
-      if (!url.url) {
-        throw new Error('Missing URL in response');
+    if (Array.isArray(options['artifact-type'])) {
+      let errors = [];
+      for (const artifactType of new Set(options['artifact-type'])) {
+        try {
+          await this.downloadArtifact(appflowId, buildId, artifactType, token, options);
+        } catch (error) {
+          errors.push(error);
+        }
       }
 
-      let customBuildFileName = '';
-
-      if (options['build-file-name']) {
-        this.env.log.warn(`The ${input('--build-file-name')} option has been deprecated. Please use ${input('--ipa-name')}, ${input('--apk-name')} or ${input('--{ artifact }-name')}.`);
-        customBuildFileName = await this.sanitizeString(options['build-file-name']);
-      } else {
-        customBuildFileName = await this.sanitizeString(options[`${artifactType.toLowerCase()}-name`]);
+      if (errors.length) {
+        errors.forEach((error) => this.env.log.error(error.message));
```

Comment:
Still need to figure out exactly how we want to handle the error.

Reply:
Probably need this to be a warning and throw an exception after that says "Some artifacts failed to download" or something similar so we can output each erroneous artifact.